cpp/documentation/_sources/_api/program_listing_file_include_vectornav_Config.hpp.rst.txt

.. _program_listing_file_include_vectornav_Config.hpp:

Program Listing for File Config.hpp
===================================

|exhale_lsh| :ref:`Return to documentation for file <file_include_vectornav_Config.hpp>` (``include\vectornav\Config.hpp``)

.. |exhale_lsh| unicode:: U+021B0 .. UPWARDS ARROW WITH TIP LEFTWARDS

.. code-block:: cpp

   #ifndef VN_CONFIG_HPP_
   #define VN_CONFIG_HPP_
   
   #include <stdint.h>
   
   #include "vectornav/HAL/Duration.hpp"
   #include "vectornav/Implementation/BinaryMeasurementDefinitions.hpp"
   #include "vectornav/Implementation/MeasurementDatatypes.hpp"
   
   namespace VN
   {
   
   #ifndef TIME_GROUP_ENABLE
   #define TIME_GROUP_ENABLE                                                                                                                                  \
       (TIME_TIMESTARTUP_BIT | TIME_TIMEGPS_BIT | TIME_TIMEGPSTOW_BIT | TIME_TIMEGPSWEEK_BIT | TIME_TIMESYNCIN_BIT | TIME_TIMEGPSPPS_BIT | TIME_TIMEUTC_BIT | \
        TIME_SYNCINCNT_BIT | TIME_SYNCOUTCNT_BIT | TIME_TIMESTATUS_BIT)
   #endif
   #ifndef IMU_GROUP_ENABLE
   #define IMU_GROUP_ENABLE                                                                                                                              \
       (IMU_IMUSTATUS_BIT | IMU_UNCOMPMAG_BIT | IMU_UNCOMPACCEL_BIT | IMU_UNCOMPGYRO_BIT | IMU_TEMPERATURE_BIT | IMU_PRESSURE_BIT | IMU_DELTATHETA_BIT | \
        IMU_DELTAVEL_BIT | IMU_MAG_BIT | IMU_ACCEL_BIT | IMU_ANGULARRATE_BIT | IMU_SENSSAT_BIT)
   #endif
   #ifndef GNSS_GROUP_ENABLE
   #define GNSS_GROUP_ENABLE                                                                                                                                      \
       (GNSS_GNSS1TIMEUTC_BIT | GNSS_GPS1TOW_BIT | GNSS_GPS1WEEK_BIT | GNSS_GNSS1NUMSATS_BIT | GNSS_GNSS1FIX_BIT | GNSS_GNSS1POSLLA_BIT | GNSS_GNSS1POSECEF_BIT | \
        GNSS_GNSS1VELNED_BIT | GNSS_GNSS1VELECEF_BIT | GNSS_GNSS1POSUNCERTAINTY_BIT | GNSS_GNSS1VELUNCERTAINTY_BIT | GNSS_GNSS1TIMEUNCERTAINTY_BIT |              \
        GNSS_GNSS1TIMEINFO_BIT | GNSS_GNSS1DOP_BIT | GNSS_GNSS1STATUS_BIT | GNSS_GNSS1ALTMSL_BIT)
   #endif
   #ifndef ATTITUDE_GROUP_ENABLE
   #define ATTITUDE_GROUP_ENABLE                                                                                                                \
       (ATTITUDE_YPR_BIT | ATTITUDE_QUATERNION_BIT | ATTITUDE_DCM_BIT | ATTITUDE_MAGNED_BIT | ATTITUDE_ACCELNED_BIT | ATTITUDE_LINBODYACC_BIT | \
        ATTITUDE_LINACCELNED_BIT | ATTITUDE_YPRU_BIT | ATTITUDE_HEAVE_BIT | ATTITUDE_ATTU_BIT)
   #endif
   #ifndef INS_GROUP_ENABLE
   #define INS_GROUP_ENABLE                                                                                                                               \
       (INS_INSSTATUS_BIT | INS_POSLLA_BIT | INS_POSECEF_BIT | INS_VELBODY_BIT | INS_VELNED_BIT | INS_VELECEF_BIT | INS_MAGECEF_BIT | INS_ACCELECEF_BIT | \
        INS_LINACCELECEF_BIT | INS_POSU_BIT | INS_VELU_BIT)
   #endif
   #ifndef GNSS2_GROUP_ENABLE
   #define GNSS2_GROUP_ENABLE                                                                                                                     \
       (GNSS2_GNSS2TIMEUTC_BIT | GNSS2_GPS2TOW_BIT | GNSS2_GPS2WEEK_BIT | GNSS2_GNSS2NUMSATS_BIT | GNSS2_GNSS2FIX_BIT | GNSS2_GNSS2POSLLA_BIT |   \
        GNSS2_GNSS2POSECEF_BIT | GNSS2_GNSS2VELNED_BIT | GNSS2_GNSS2VELECEF_BIT | GNSS2_GNSS2POSUNCERTAINTY_BIT | GNSS2_GNSS2VELUNCERTAINTY_BIT | \
        GNSS2_GNSS2TIMEUNCERTAINTY_BIT | GNSS2_GNSS2TIMEINFO_BIT | GNSS2_GNSS2DOP_BIT | GNSS2_GNSS2STATUS_BIT | GNSS2_GNSS2ALTMSL_BIT)
   #endif
   #ifndef GNSS3_GROUP_ENABLE
   #define GNSS3_GROUP_ENABLE 0
   #endif
   
   #ifndef THREADING_ENABLE
   #define THREADING_ENABLE true
   #endif
   
   namespace Config
   {
   
   namespace PacketFinders
   {
   // Universal
   constexpr uint64_t mainBufferCapacity = 4096;
   constexpr uint64_t packetMaxLength = 3000;
   constexpr uint8_t maxNumPacketFinders = 3;  // FA , Ascii and FB
   
   // Fa
   constexpr uint16_t faPacketMaxLength = 2000;
   constexpr uint8_t gnssSatInfoMaxCount = GNSS_SAT_INFO_MAX_COUNT;  // Defiend in MeasurementDatatypes.hpp to avoid circular dependency
   constexpr uint8_t gnssRawMeasMaxCount = GNSS_RAW_MEAS_MAX_COUNT;  // Defiend in MeasurementDatatypes.hpp to avoid circular dependency
   
   // Ascii
   constexpr uint8_t asciiMaxFieldCount = 40;
   constexpr uint16_t asciiHeaderMaxLength = 10;
   constexpr uint16_t asciiPacketMaxLength = 320;
   constexpr uint8_t asciiFieldMaxLength = 20;
   
   // Fb
   constexpr uint16_t fbPacketMaxLength = 2000;
   constexpr uint64_t fbBufferCapacity = fbPacketMaxLength * 2;
   }  // namespace PacketFinders
   
   namespace PacketDispatchers
   {
   // Universal
   constexpr EnabledMeasurements cdEnabledMeasTypes = {
       TIME_GROUP_ENABLE, IMU_GROUP_ENABLE, GNSS_GROUP_ENABLE, ATTITUDE_GROUP_ENABLE, INS_GROUP_ENABLE, GNSS2_GROUP_ENABLE, 0, 0, 0, 0, 0, GNSS3_GROUP_ENABLE};
   constexpr uint8_t compositeDataQueueCapacity = 100;
   
   // Fa
   constexpr uint8_t faPacketSubscriberCapacity = 5;
   
   // Ascii
   constexpr uint8_t asciiPacketSubscriberCapacity = 5;
   }  // namespace PacketDispatchers
   
   namespace Serial
   {
   constexpr uint16_t PortNameMaxLength = 32;
   }  // namespace Serial
   
   namespace Sensor
   {
   // Timers
   constexpr Microseconds commandSendTimeoutLength = 100ms;
   constexpr Microseconds wnvSendTimeoutLength = 1200ms;
   constexpr Microseconds getMeasurementTimeoutLength = 100ms;
   
   // Sleeps
   constexpr Microseconds resetSleepDuration = 2500ms;
   constexpr Microseconds listenSleepDuration = 1ms;
   constexpr Microseconds getMeasurementSleepDuration = 100us;
   constexpr Microseconds commandSendSleepDuration = 100us;
   
   // Retries
   constexpr uint8_t commandSendRetriesAllowed = 2;
   constexpr bool retryVerifyConnectivity = true;
   }  // namespace Sensor
   
   namespace CommandProcessor
   {
   constexpr Microseconds commandRemovalTimeoutLength = Sensor::commandSendTimeoutLength * 2;
   constexpr uint8_t commandProcQueueCapacity = 10;
   constexpr size_t messageMaxLength = 350;
   }  // namespace CommandProcessor
   
   namespace Errors
   {
   constexpr uint8_t asyncErrorQueueCapacity = 5;
   constexpr uint16_t asyncErrorMessageCapacity = 256;
   }  // namespace Errors
   
   // Common sense checks
   static_assert(CommandProcessor::messageMaxLength >= PacketFinders::asciiPacketMaxLength);  // No use to find a packet larger than we can propagate
   static_assert(PacketFinders::asciiPacketMaxLength > PacketFinders::asciiFieldMaxLength);
   static_assert(PacketFinders::asciiPacketMaxLength > PacketFinders::asciiHeaderMaxLength);
   static_assert(PacketFinders::mainBufferCapacity >= PacketFinders::packetMaxLength);
   static_assert(PacketFinders::packetMaxLength > CommandProcessor::messageMaxLength);
   static_assert(PacketFinders::packetMaxLength > PacketFinders::faPacketMaxLength);
   static_assert(PacketFinders::packetMaxLength > PacketFinders::fbPacketMaxLength);
   
   }  // namespace Config
   
   }  // namespace VN
   
   #endif  // VN_CONFIG_HPP_


net/documentation/_sources/_api/class_v_n_s_d_k_1_1_registers_1_1_g_n_s_s_1_1_gnss_system_config.rst.txt
.. _exhale_class_class_v_n_s_d_k_1_1_registers_1_1_g_n_s_s_1_1_gnss_system_config:

Class GnssSystemConfig
======================

- Defined in :ref:`file_documentation__include_CLI_Registers.hpp`


Nested Relationships
--------------------


Nested Types
************



Inheritance Relationships
-------------------------

Base Type
*********

- ``public VNSDK::Registers::Register`` (:ref:`exhale_class_class_v_n_s_d_k_1_1_registers_1_1_register`)


Class Documentation
-------------------


.. doxygenclass:: VNSDK::Registers::GNSS::GnssSystemConfig
   :project: VectorNav SDK
   :members:
   :protected-members:
   :undoc-members:

changelog.txt
0.99.0
- Global  : Switched SkippedByte subscription from a ByteBuffer to a PacketQueue
             - Removed registerSkippedByteBuffer and deregisterSkippedByteBuffer methods from Sensor
             - Overloaded subscribeToMessage to allow passing only a SyncByte to subscribe to messages 
               (including SyncByte::None for SkippedBytes)
             - Created a new defaultMetadata option for PacketDetails for constructing SkippedByte packets
             - Updated DataExport plugin to use new ExporterSkippedByte class based on queue
 - Global  : Made ConfigurationRegister fields std::optional, requiring all to be set to write to register
             - toWriteCommand() method to returns optional/nullable to allow for failure to generate string
             - C++ bitfields must be first instantiated with emplace() before setting the bitfields
 - Global  : Renamed Sensor::asynchronousErrorQueueSize and Sensor::getAsynchronousError to
             asyncErrorQueueSize and getNextAsyncError for consistency
 - Global  : Added ability for Sensor to 'connect' to a file instead of a com port for file processing
             - Introduced a new MeasQueueMode to Sensor constructor to allow the CompositeData measurement
               queue to be disabled or set to Force/Try/Retry push modes (default to Force, backwards 
               compatible)
             - Updated ByteBuffer for improved thread-safety and added new methods to allow inputting to 
               ByteBuffer from a file.  Renamed numLinearBytes() to numLinearBytesToPeek().
             - Updated DirectAccessQueue for Force/Try/Retry put modes
             - Updated DirectAccessQueue to only erase a single entry when Force putting instead of 
               erasing all
             - Changed InputFile.read() to return number of bytes read instead of Errored
 - Global  : New Error::AlreadyConnected returned if connect() or autoconnect() called when sensor already
             connected to either file or com port (previously succeeded at second connection)
 - Global  : Added ability to Subscribe to raw FbPackets (formerly SplitPackets)
 - Cpp     : Added vectornav sublayer for Logger plugin
 - Cpp     : Implemented packet length check when pushing to a PacketQueue to fix buffer overflow bug.  If a 
             packet exceeds PacketQueue message capacity, then a Packet of SyncByte::None and length=0 put on
             queue, and Error::PacketQueueOverrun sent to asyncErrorQueue.
 - Cpp     : Sensor function subscribeToMessage() for ASCII and FA packets now requires a filter argument 
             (instead of optional).  Can use new overload requiring only SyncByte to achieve same result.
 - Cpp     : Moved receivedByteBuffer loads from PacketSynchronizer to Sensor.  Eliminated 
             skippedReceivedByteBufferMaxPutLength parameter and associated temporary buffer.
 - Cpp     : Updated Windows and Linux Serial drivers to eliminate buffer overflow bug.  Removed intermediate
             buffers from each, eliminating numBytesToReadPerGetData parameter in Config.hpp.
 - Cpp     : Added maxPacketLength parameter to Config.hpp as a sanity check in PacketSynchronizer across all 
             packet types  
 - Cpp     : Added Sensor mutex for subscribing or unsubscribing from messages
 - Cpp     : Introduced VN_DEBUG_0 to replace cout/cerr in plugins. Messages can be disabled by setting
             VN_DEBUG_LEVEL to -1 in Debug.hpp
 - Cpp     : Removed most VN_ABORT() and several VN_ASSERT(), returning error codes instead
 - Cpp     : Scrubbed code of references to STL that could throw exceptions or dynamically allocate
 - Cpp     : BinaryHeader::toMeasurementHeader now returns std::optional to allow for failure
 - Cpp     : Removed custom Error definition in CommandProcessor
 - Cpp     : Introduced Errored alias for bool return type for clarity
 - Cpp     : Introduced ErrorAll type that combines standard Error with errors defined by enabled plugins
 - Cpp     : Changed addSubscriber methods to return Error instead of bool
 - Cpp     : Renamed GenericCommand::matchResponse to isMatchingResponse for consistency
 - Cpp     : Changed FaPacketProtocol Metadata member 'length' from size_t to uint16_t
 - Cpp     : Updated FbPacket handling, including renaming from SplitPacket and adding subscribers
 - Cpp     : Renamed Packet member 'size' to 'capacity'
 - Cpp     : Fixed bug in removeSubscriber methods
 - Cpp     : Updated restoreFactorySettings command to return immediately if an async packet is received
 - Cpp     : Renamed CalculateCheckSum and CalculateCRC to calculateChecksum and calculateCRC for
             consistency
 - Cpp     : Added plugin namespaces
 - Cpp     : Added AsyncError throwing to the Packet Dispatchers
 - Cpp     : In Windows' serial driver, add invalid path to count as an InvalidPortName rather than
             unexpected
 - Cpp     : Move register's operator== definition to header
 - Cpp     : Alias Register::*::InsStatus from InsStatus directly
 - Cpp     : Add bitfield equality and inequality operators
 - Cpp     : Add padding bits to the end of Register bitfields to ensure packing uniformity
 - Cpp     : Add BinaryHeader::toHeaderBytes to convert to a vector of bytes, for comparison with an
             unparsed packet
 - Cpp     : Add BinaryOutputMeasurements::toHeaderBytes
 - Cpp     : Added a PacketMetadata template struct to PacketDispatcher.hpp as a base for all protocols
 - Python  : Renamed getNextAsyncError to throwIfAsyncError which now throws instead of returning error
 - Python  : Updated RegisterScan plugin to raise exceptions when errors are returned
 - Python  : Updated all examples to use try-except on library calls that can raise exceptions
 - Python  : Wrap toHeaderBytes into BinaryOutputMeasurements::toHeaderBytes

cpp/include/vectornav/Config.hpp
// The MIT License (MIT)
// 
// VectorNav SDK (v0.99.0)
// Copyright (c) 2024 VectorNav Technologies, LLC
// 
// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the "Software"), to deal
// in the Software without restriction, including without limitation the rights
// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
// copies of the Software, and to permit persons to whom the Software is
// furnished to do so, subject to the following conditions:
// 
// The above copyright notice and this permission notice shall be included in
// all copies or substantial portions of the Software.
// 
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
// THE SOFTWARE.

#ifndef VN_CONFIG_HPP_
#define VN_CONFIG_HPP_

#include <stdint.h>

#include "vectornav/HAL/Duration.hpp"
#include "vectornav/Implementation/BinaryMeasurementDefinitions.hpp"
#include "vectornav/Implementation/MeasurementDatatypes.hpp"

namespace VN
{

#ifndef TIME_GROUP_ENABLE
#define TIME_GROUP_ENABLE                                                                                                                                  \
    (TIME_TIMESTARTUP_BIT | TIME_TIMEGPS_BIT | TIME_TIMEGPSTOW_BIT | TIME_TIMEGPSWEEK_BIT | TIME_TIMESYNCIN_BIT | TIME_TIMEGPSPPS_BIT | TIME_TIMEUTC_BIT | \
     TIME_SYNCINCNT_BIT | TIME_SYNCOUTCNT_BIT | TIME_TIMESTATUS_BIT)
#endif
#ifndef IMU_GROUP_ENABLE
#define IMU_GROUP_ENABLE                                                                                                                              \
    (IMU_IMUSTATUS_BIT | IMU_UNCOMPMAG_BIT | IMU_UNCOMPACCEL_BIT | IMU_UNCOMPGYRO_BIT | IMU_TEMPERATURE_BIT | IMU_PRESSURE_BIT | IMU_DELTATHETA_BIT | \
     IMU_DELTAVEL_BIT | IMU_MAG_BIT | IMU_ACCEL_BIT | IMU_ANGULARRATE_BIT | IMU_SENSSAT_BIT)
#endif
#ifndef GNSS_GROUP_ENABLE
#define GNSS_GROUP_ENABLE                                                                                                                                      \
    (GNSS_GNSS1TIMEUTC_BIT | GNSS_GPS1TOW_BIT | GNSS_GPS1WEEK_BIT | GNSS_GNSS1NUMSATS_BIT | GNSS_GNSS1FIX_BIT | GNSS_GNSS1POSLLA_BIT | GNSS_GNSS1POSECEF_BIT | \
     GNSS_GNSS1VELNED_BIT | GNSS_GNSS1VELECEF_BIT | GNSS_GNSS1POSUNCERTAINTY_BIT | GNSS_GNSS1VELUNCERTAINTY_BIT | GNSS_GNSS1TIMEUNCERTAINTY_BIT |              \
     GNSS_GNSS1TIMEINFO_BIT | GNSS_GNSS1DOP_BIT | GNSS_GNSS1STATUS_BIT | GNSS_GNSS1ALTMSL_BIT)
#endif
#ifndef ATTITUDE_GROUP_ENABLE
#define ATTITUDE_GROUP_ENABLE                                                                                                                \
    (ATTITUDE_YPR_BIT | ATTITUDE_QUATERNION_BIT | ATTITUDE_DCM_BIT | ATTITUDE_MAGNED_BIT | ATTITUDE_ACCELNED_BIT | ATTITUDE_LINBODYACC_BIT | \
     ATTITUDE_LINACCELNED_BIT | ATTITUDE_YPRU_BIT | ATTITUDE_HEAVE_BIT | ATTITUDE_ATTU_BIT)
#endif
#ifndef INS_GROUP_ENABLE
#define INS_GROUP_ENABLE                                                                                                                               \
    (INS_INSSTATUS_BIT | INS_POSLLA_BIT | INS_POSECEF_BIT | INS_VELBODY_BIT | INS_VELNED_BIT | INS_VELECEF_BIT | INS_MAGECEF_BIT | INS_ACCELECEF_BIT | \
     INS_LINACCELECEF_BIT | INS_POSU_BIT | INS_VELU_BIT)
#endif
#ifndef GNSS2_GROUP_ENABLE
#define GNSS2_GROUP_ENABLE                                                                                                                     \
    (GNSS2_GNSS2TIMEUTC_BIT | GNSS2_GPS2TOW_BIT | GNSS2_GPS2WEEK_BIT | GNSS2_GNSS2NUMSATS_BIT | GNSS2_GNSS2FIX_BIT | GNSS2_GNSS2POSLLA_BIT |   \
     GNSS2_GNSS2POSECEF_BIT | GNSS2_GNSS2VELNED_BIT | GNSS2_GNSS2VELECEF_BIT | GNSS2_GNSS2POSUNCERTAINTY_BIT | GNSS2_GNSS2VELUNCERTAINTY_BIT | \
     GNSS2_GNSS2TIMEUNCERTAINTY_BIT | GNSS2_GNSS2TIMEINFO_BIT | GNSS2_GNSS2DOP_BIT | GNSS2_GNSS2STATUS_BIT | GNSS2_GNSS2ALTMSL_BIT)
#endif
#ifndef GNSS3_GROUP_ENABLE
#define GNSS3_GROUP_ENABLE 0
#endif

#ifndef THREADING_ENABLE
#define THREADING_ENABLE true
#endif

namespace Config
{

namespace PacketFinders
{
// Universal
constexpr uint64_t mainBufferCapacity = 4096;
constexpr uint64_t packetMaxLength = 3000;
constexpr uint8_t maxNumPacketFinders = 3;  // FA , Ascii and FB

// Fa
constexpr uint16_t faPacketMaxLength = 2000;
constexpr uint8_t gnssSatInfoMaxCount = GNSS_SAT_INFO_MAX_COUNT;  // Defiend in MeasurementDatatypes.hpp to avoid circular dependency
constexpr uint8_t gnssRawMeasMaxCount = GNSS_RAW_MEAS_MAX_COUNT;  // Defiend in MeasurementDatatypes.hpp to avoid circular dependency

// Ascii
constexpr uint8_t asciiMaxFieldCount = 40;
constexpr uint16_t asciiHeaderMaxLength = 10;
constexpr uint16_t asciiPacketMaxLength = 320;
constexpr uint8_t asciiFieldMaxLength = 20;

// Fb
constexpr uint16_t fbPacketMaxLength = 2000;
constexpr uint64_t fbBufferCapacity = fbPacketMaxLength * 2;
}  // namespace PacketFinders

namespace PacketDispatchers
{
// Universal
constexpr EnabledMeasurements cdEnabledMeasTypes = {
    TIME_GROUP_ENABLE, IMU_GROUP_ENABLE, GNSS_GROUP_ENABLE, ATTITUDE_GROUP_ENABLE, INS_GROUP_ENABLE, GNSS2_GROUP_ENABLE, 0, 0, 0, 0, 0, GNSS3_GROUP_ENABLE};
constexpr uint8_t compositeDataQueueCapacity = 100;

// Fa
constexpr uint8_t faPacketSubscriberCapacity = 5;

// Ascii
constexpr uint8_t asciiPacketSubscriberCapacity = 5;
}  // namespace PacketDispatchers

namespace Serial
{
constexpr uint16_t PortNameMaxLength = 4096;
}  // namespace Serial

namespace Sensor
{
// Timers
constexpr Microseconds commandSendTimeoutLength = 100ms;
constexpr Microseconds wnvSendTimeoutLength = 1200ms;
constexpr Microseconds getMeasurementTimeoutLength = 100ms;

// Sleeps
constexpr Microseconds resetSleepDuration = 2500ms;
constexpr Microseconds listenSleepDuration = 1ms;
constexpr Microseconds getMeasurementSleepDuration = 100us;
constexpr Microseconds commandSendSleepDuration = 100us;

// Retries
constexpr uint8_t commandSendRetriesAllowed = 2;
constexpr bool retryVerifyConnectivity = true;
}  // namespace Sensor

namespace CommandProcessor
{
constexpr Microseconds commandRemovalTimeoutLength = Sensor::commandSendTimeoutLength * 2;
constexpr uint8_t commandProcQueueCapacity = 10;
constexpr size_t messageMaxLength = 350;
}  // namespace CommandProcessor

namespace Errors
{
constexpr uint8_t asyncErrorQueueCapacity = 5;
constexpr uint16_t asyncErrorMessageCapacity = 256;
}  // namespace Errors

// Common sense checks
static_assert(CommandProcessor::messageMaxLength >= PacketFinders::asciiPacketMaxLength);  // No use to find a packet larger than we can propagate
static_assert(PacketFinders::asciiPacketMaxLength > PacketFinders::asciiFieldMaxLength);
static_assert(PacketFinders::asciiPacketMaxLength > PacketFinders::asciiHeaderMaxLength);
static_assert(PacketFinders::mainBufferCapacity >= PacketFinders::packetMaxLength);
static_assert(PacketFinders::packetMaxLength > CommandProcessor::messageMaxLength);
static_assert(PacketFinders::packetMaxLength > PacketFinders::faPacketMaxLength);
static_assert(PacketFinders::packetMaxLength > PacketFinders::fbPacketMaxLength);

}  // namespace Config

}  // namespace VN

#endif  // VN_CONFIG_HPP_
